signature/revocation.go

```diff
@@ -0,0 +1,26 @@
+package signature
+
+import (
+	"crypto/x509"
+	"net/http"
+
+	"github.com/cloudflare/cfssl/revoke"
+)
+
+type RevocationChecker struct {
+	HTTPClient *http.Client
+}
+
+func NewRevocationChecker(httpClients ...http.Client) *RevocationChecker {
+	if len(httpClients) == 0 {
+		return &RevocationChecker{HTTPClient: http.DefaultClient}
+	} else {
+		return &RevocationChecker{HTTPClient: &httpClients[0]}
+	}
+}
+
+func (r RevocationChecker) CheckRevocationStatus(cert *x509.Certificate) (bool, error) {
+	revoke.HTTPClient = r.HTTPClient
+	revoked, _, err := revoke.VerifyCertificateError(cert)
```

Comment:
This method does more than revocation check which we dont want
https://github.com/cloudflare/cfssl/blob/master/revoke/revoke.go#L191-L202